### PR TITLE
Add series attribute to bundles

### PR DESCRIPTION
**Pull request by wallyworld**
Bundle allow a series to be specified. It is an error to specify a series and a charm store charm. Series only applies to local charms.

### CODE REVIEW

File: bundledata.go [hunk 1]
```diff
@@ -68,6 +68,11 @@ type ServiceSpec struct {
 	// use for the given service.
 	Charm string
 
+	// Series is the series to use when deploying the charm
```
Comment (axw) [inline bundledata.go]: worth mentioning here that the series may not be specified for cs: charms?

File: bundledata.go [hunk 2]
```diff
@@ -364,9 +371,18 @@ func (verifier *bundleDataVerifier) verifyServices() {
...
 				}
 			}
-		} else if _, err := ParseURL(svc.Charm); err != nil {
+		} else if curl, err = ParseURL(svc.Charm); err != nil {
 			verifier.addErrorf("invalid charm URL in service %q: %v", name, err)
 		}
+
+		// Check the series.
+		if curl != nil && svc.Series != "" {
+			verifier.addErrorf("service %q declares both a series and a charm store URL", name)
```
Comment (axw) [inline bundledata.go]: "store" in the message may not be accurate in the future, if we add additional URL schemes

**Comment by axw** [comment]
LGTM with a couple of quibbles

File: bundledata.go (hunk 1 above)
Comment (wallyworld) [inline bundledata.go]: done

File: bundledata.go (hunk 2 above)
Comment (wallyworld) [inline bundledata.go]: done

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**